packages/utils/test/mergeDefaultsWithFormData.test.ts

```diff
@@ -307,4 +307,94 @@ describe('mergeDefaultsWithFormData()', () => {
       expect(mergeDefaultsWithFormData(obj1, obj2)?.a).toBeInstanceOf(File);
     });
   });
+
+  describe('test with overrideFormDataWithDefaults set to `replace`', () => {
+    it('should return empty array even when no defaults', () => {
+      expect(mergeDefaultsWithFormData(undefined, [2], undefined, undefined, 'replace')).toEqual([]);
+    });
+
+    it('should return default when formData is undefined', () => {
+      expect(mergeDefaultsWithFormData({}, undefined, undefined, undefined, 'replace')).toEqual({});
+    });
+
+    it('should return default when formData is undefined and defaultSupercedesUndefined true', () => {
+      expect(mergeDefaultsWithFormData({}, undefined, undefined, true, 'replace')).toEqual({});
+    });
+
+    it('should return default when formData is null and defaultSupercedesUndefined true', () => {
+      expect(mergeDefaultsWithFormData({}, null, undefined, true, 'replace')).toEqual({});
+    });
+
+    it('should not merge two one-level deep objects', () => {
+      expect(mergeDefaultsWithFormData({ a: 1 }, { b: 2 }, undefined, undefined, 'replace')).toEqual({
+        a: 1,
+      });
+    });
+
+    it('should not override the first object with the values from the second', () => {
+      expect(mergeDefaultsWithFormData({ a: 1 }, { a: 2 }, undefined, undefined, 'replace')).toEqual({ a: 1 });
+    });
+
+    it('should not return undefined from defaults', () => {
+      expect(
+        mergeDefaultsWithFormData({ a: { b: undefined } }, { a: { b: { c: 1 } } }, undefined, undefined, 'replace'),
+      ).toEqual({
+        a: { b: {} },
+      });
+    });
```

Comment:
`undefined` is not a valid JSON value,. so what JSON Schema results in a `default` value that looks like `{ a: { b: undefined } }`? Are we doing something special in RJSF that I am ignorant of?

Reply:
I do not know actually how it might happen in JSON schema, I just tried to repeat existing tests. Here we wiill replace missing value (undefined from default arg) with proper default for object (empty one). Do you think it's incorect?

Reply:
It looks like our `computeDefaults` utility may return `undefined`, so I guess we do need to handle it.